lambda/import/index.js

```diff
@@ -1,22 +1,73 @@
-var Promise = require('bluebird')
-var aws = require("aws-sdk")
-aws.config.setPromisesDependency(Promise)
-aws.config.region = process.env.AWS_REGION
+var Promise = require('bluebird');
+var aws = require("aws-sdk");
+aws.config.setPromisesDependency(Promise);
+aws.config.region = process.env.AWS_REGION;
+const get_embeddings = require('./embeddings');
 
-var s3 = new aws.S3()
-var lambda = new aws.Lambda()
-var stride = parseInt(process.env.STRIDE)
-var _ = require('lodash')
-var convertxlsx = require('convert-xlsx')
-var delete_existing_content = require('delete_existing_content')
+var s3 = new aws.S3();
+var lambda = new aws.Lambda();
+var stride = parseInt(process.env.STRIDE);
+var _ = require('lodash');
+var convertxlsx = require('convert-xlsx');
+var delete_existing_content = require('delete_existing_content');
 
+function isJson(str) {
+    try {
+        JSON.parse(str);
+    } catch (e) {
+        return false;
+    }
+    return true;
+}
+
+function str2bool(settings) {
+    var new_settings = _.mapValues(settings, x => {
+        if (_.isString(x)) {
+            x = x.replace(/^"(.+)"$/,'$1');  // remove wrapping quotes
+            if (x.toLowerCase() === "true") {
+                return true ;
+            }
+            if (x.toLowerCase() === "false") {
+                return false ;
+            }
+        }
+        return x;
+    });
+    return new_settings;
+}
+
+async function get_parameter(param_name) {
+    var ssm = new aws.SSM();
+    var params = {
+        Name: param_name,
+        WithDecryption: true
+    };
+    var response = await ssm.getParameter(params).promise();
+    var settings = response.Parameter.Value
+    if (isJson(settings)) {
+        settings = JSON.parse(response.Parameter.Value);
+        settings = str2bool(settings) ;
+    }
+    return settings;
+}
+
+const get_settings = async function get_settings() {
+    var default_settings_param = process.env.DEFAULT_SETTINGS_PARAM;
+    var custom_settings_param = process.env.CUSTOM_SETTINGS_PARAM;
+    console.log("Getting Default QnABot settings from SSM Parameter Store: ", default_settings_param);
+    var default_settings = await get_parameter(default_settings_param);
+    console.log("Getting Custom QnABot settings from SSM Parameter Store: ", custom_settings_param);
+    var custom_settings = await get_parameter(custom_settings_param);
+    var settings = _.merge(default_settings, custom_settings);
+    console.log("Merged Settings: ", settings);
+    return settings;
+}
```

Comment:
@rstrahan all of this functionality already exists in the qnabot-common-layer. Similar to the previous comment, can we add es-proxy-layer and qnabot-common-layer to the Import Lambdas?

Reply:
Done